server/rest_slicer_cli.py

```diff
@@ -826,14 +829,163 @@ def genRESTEndPointsForSlicerCLIsInDocker(info, restResource, dockerImages):
     info
     restResource : str or girder.api.rest.Resource
         REST resource to which the end-points should be attached
-    dockerImages : str or list of str
-        A single docker image or a list of docker images
+    dockerImages : a list of docker image names
+
 
     Returns
     -------
 
     """
+    dockerImages
+    # validate restResource argument
+    if not isinstance(restResource, (str, Resource)):
+        raise Exception('restResource must either be a string or '
+                        'an object of girder.api.rest.Resource')
+
+    # validate dockerImages arguments
+    if not isinstance(dockerImages, (str, list)):
+        raise Exception('dockerImages must either be a single docker image '
+                        'string or a list of docker image strings')
+
+    if isinstance(dockerImages, list):
+        for img in dockerImages:
+            if not isinstance(img, str):
+                raise Exception('dockerImages must either be a single '
+                                'docker image string or a list of docker '
+                                'image strings')
+    else:
+        dockerImages = [dockerImages]
+
+    # create REST resource if given a name
+    if isinstance(restResource, str):
+        restResource = type(restResource,
+                            (Resource, ),
+                            {'resourceName': restResource})()
+
+    restResourceName = type(restResource).__name__
+
+    # Add REST routes for slicer CLIs in each docker image
+    cliList = []
+
+    for dimg in dockerImages:
+        # check if the docker image exists
+
+        id = localDockerImageExists(dimg)
+        if id is None:
+            id = pullDockerImage(dimg)
+        if id is None:
+            raise ValueError("The image %s does not exist locally or "
+                             "on the default registry" % dimg)
```

Comment:
Lines 873-878 that check if the docker image exists locally and tried to pull if not available can be wrapped into a function say `getDockerImage(image_string, pull=True)`